### PR TITLE
Add persistent others to dock

### PR DESCRIPTION
**Pull request by rmgpinto**
Add persistent others to dock. Closes https://github.com/LnL7/nix-darwin/issues/927.

### CODE REVIEW

**Comment by rmgpinto** [comment]
@Enzime @Samasaur1

**Review by Samasaur1 — CHANGES_REQUESTED** [review]
The actual change here looks good to me, but I'd like it if you could avoid making unrelated formatting-only changes in the same PR

**Comment by rmgpinto** [comment]
@Samasaur1  I didn't notice, it's my linter and auto format on save. I've reverted the formatting changes.

**Comment by rmgpinto** [comment]
@Samasaur1 I can't seem to understand why the tests fail. Are you able to help?

**Comment by Samasaur1** [comment]
Okay I manually checked the failing test on unstable, and it's only failing because of a missing newline. Seems like a false alarm; let me take a closer look and determine whether it's better to change the test or change the implementation back

**Comment by Samasaur1** [comment]
Ah, okay, you need another newline in the test file

File: tests/fixtures/system-defaults-write/activate-user.txt [hunk 1]
```diff
@@ -255,7 +255,36 @@ defaults write com.apple.dock 'persistent-apps' $'<?xml version="1.0" encoding="
...
+			<dict>
+				<key>_CFURLString</key>
+				<string>~/Downloads</string>
+				<key>_CFURLStringType</key>
+				<integer>15</integer>
+			</dict>
+		</dict>
+	</dict>
+</array>
+</plist>'
```
Comment (Samasaur1) [inline tests/fixtures/system-defaults-write/activate-user.txt:288]:
```suggestion


```

**Review by Samasaur1 — CHANGES_REQUESTED** [review]

**Comment by rmgpinto** [comment]
Thanks @Samasaur1, tests passed.

**Comment by Enzime** [comment]
Can you rebase the PR?

**Comment by rmgpinto** [comment]
done @Enzime

**Comment by Enzime** [comment]
Can you rebase your PR on top of the latest master and squash your commits? Ideally your branch shouldn't contain the `fmt` commits and should only contain your commits

**Comment by rmgpinto** [comment]
should be ok now @Enzime

**Comment by rmgpinto** [comment]
The tests seem to be flaky, sometimes that check passes, others it doesn't.

**Review by Enzime — APPROVED** [review]
LGTM

Thanks for the contribution

**Comment by thegleich** [comment]
This introduces a regression (warning) by using `lib.mdDoc`

`mdDoc` usage has been been removed in https://github.com/LnL7/nix-darwin/pull/932

**Comment by rmgpinto** [comment]
Fixing in https://github.com/LnL7/nix-darwin/pull/950

**Pull request merged**